great_expectations/datasource/fluent/interfaces.py

```diff
@@ -463,10 +462,6 @@ def add_asset(self, asset: _DataAssetT) -> _DataAssetT:
 
         self.assets[asset.name] = asset
 
-        # pydantic needs to know that an asset has been set so that it doesn't get excluded
-        # when dumping to dict, json, yaml etc.
-        self.__fields_set__.update(_FIELDS_ALWAYS_SET)
```

Comment:
Why don't we need this anymore?

Reply:
Because we do this same logic before serialization. It's very likely we didn't actually need this before.